reflectutil/setter.go

```diff
@@ -2,51 +2,104 @@ package reflectutil
 
 import (
 	"fmt"
+	"go/ast"
 	"reflect"
 	"strconv"
 	"strings"
 )
 
-// SetValue assign value to the field
-// name of the struct structPtr.
+type protoLiteral interface {
+	SourceRepresentation() string
+}
+
+// SetValue assign value to the field name of the struct structPtr.
 // Returns an error if name is not found or structPtr cannot
 // be addressed.
-func SetValue(structPtr interface{}, name, value string) {
+func SetValue(structPtr interface{}, name, value interface{}) {
+	nameStr := ""
+	switch name := name.(type) {
+	case *ast.Ident:
+		nameStr = name.Name
+	case string:
+		nameStr = name
+	default:
+		panic(fmt.Errorf("invalid name type: %T", name))
+	}
+
 	// Performs a case insensitive search because generated structs by
 	// protoc don't follow the best practices from the go naming convention:
-	// initialisms should be all capitals
-	tmp := reflect.Indirect(reflect.ValueOf(structPtr)).FieldByNameFunc(func(s string) bool {
-		return strings.EqualFold(s, name)
+	// initialisms should be all capitals.
+	// Remove underscores too, so that foo_bar matches the field FooBar.
+	nameStr = strings.ReplaceAll(nameStr, "_", "")
```

Comment:
I think this is why travis fails

Reply:
Let me drop 1.11 support ;) after all, 1.13 is around the corner, and 1.11 has somewhat bad module support.